src/test/java/it/gov/pagopa/pu/bff/connector/AuthClientImplTest.java

```diff
@@ -25,27 +25,27 @@
 class AuthClientImplTest {
 
   @Mock
-  private RestTemplateBuilder restTemplateBuilder;
+  private RestTemplateBuilder restTemplateBuilderMock;
 
   @Mock
-  private AuthnApi authApi;
+  private AuthnApi authApiMock;
 
   @InjectMocks
-  private AuthClientImpl authClient;
+  private AuthClientImpl authClientMock;
 
   @Mock
-  RestTemplate restTemplate;
+  private RestTemplate restTemplateMock;
 
   private String baseUrl = "http://example.com"; // Set a default value for testing
 
   @BeforeEach
   void setUp() {
-    restTemplate = Mockito.mock(RestTemplate.class);
-    Mockito.when(restTemplateBuilder.build()).thenReturn(restTemplate);
-    Mockito.when(restTemplate.getUriTemplateHandler()).thenReturn(new DefaultUriBuilderFactory());
-    ApiClient apiClient = new ApiClient(restTemplate);
+    restTemplateMock = Mockito.mock(RestTemplate.class);
```

Comment:
if you use MockitoExtension, and @Mock annotation, you don't need to create again the mock here

Reply:
removed